src/bootstrap-table.js

```diff
@@ -503,78 +503,84 @@ class BootstrapTable {
       o.icons = Utils.calculateObjectValue(null, o.icons)
     }
 
-    if (o.showPaginationSwitch) {
-      html.push(`<button class="${this.constants.buttonsClass}" type="button" name="paginationSwitch"
+    const buttonsHtml = {
+      paginationSwitch: `<button class="${this.constants.buttonsClass}" type="button" name="paginationSwitch"
         aria-label="Pagination Switch" title="${o.formatPaginationSwitch()}">
         ${o.showButtonIcons ? Utils.sprintf(this.constants.html.icon, o.iconsPrefix, o.icons.paginationSwitchDown) : ''}
         ${o.showButtonText ? o.formatPaginationSwitchUp() : ''}
-        </button>`)
-    }
+        </button>`,
 
-    if (o.showRefresh) {
-      html.push(`<button class="${this.constants.buttonsClass}" type="button" name="refresh"
+      refresh: `<button class="${this.constants.buttonsClass}" type="button" name="refresh"
         aria-label="Refresh" title="${o.formatRefresh()}">
         ${o.showButtonIcons ? Utils.sprintf(this.constants.html.icon, o.iconsPrefix, o.icons.refresh) : ''}
         ${o.showButtonText ? o.formatRefresh() : ''}
-        </button>`)
-    }
+        </button>`,
 
-    if (o.showToggle) {
-      html.push(`<button class="${this.constants.buttonsClass}" type="button" name="toggle"
+      toggle: `<button class="${this.constants.buttonsClass}" type="button" name="toggle"
         aria-label="Toggle" title="${o.formatToggle()}">
         ${o.showButtonIcons ? Utils.sprintf(this.constants.html.icon, o.iconsPrefix, o.icons.toggleOff) : '' }
         ${o.showButtonText ? o.formatToggleOn() : ''}
-        </button>`)
-    }
+        </button>`,
 
-    if (o.showFullscreen) {
-      html.push(`<button class="${this.constants.buttonsClass}" type="button" name="fullscreen"
+      fullscreen: `<button class="${this.constants.buttonsClass}" type="button" name="fullscreen"
         aria-label="Fullscreen" title="${o.formatFullscreen()}">
         ${o.showButtonIcons ? Utils.sprintf(this.constants.html.icon, o.iconsPrefix, o.icons.fullscreen) : '' }
         ${o.showButtonText ? o.formatFullscreen() : ''}
```

Comment:
I know this is not part of this PR, but can we change the `o`variable name to something more meaningful like `opts`?

Reply:
Done!